tests/test_unit.py

```diff
@@ -115,6 +115,41 @@ def test_normalize(self):
             )
             df_norm = df_utils.normalize(df, data_params)
 
+    def test_add_lagged_regressors(self):
+        NROWS = 512
+        EPOCHS = 3
+        BATCH_SIZE = 32
+        
+        df = pd.read_csv(PEYTON_FILE, nrows=NROWS)
+
+        df["A"] = df["y"].rolling(7, min_periods=1).mean()
+        df["B"] = df["y"].rolling(15, min_periods=1).mean()
+        df["C"] = df["y"].rolling(30, min_periods=1).mean()
+        
+        check = {
+            "1": ["A"],
+            "2": ["A", "B"],
+            "3": ["A", "B", "C"],
+        }
+        
+        for key, value in check.items():
+            print(value)    
+            feats=np.array(['ds','y']+value)
+            df1 = pd.DataFrame(df, columns=feats)
+            cols = [col for col in df1.columns if col not in ['ds','y']]
+            m = NeuralProphet(
+                    n_forecasts=7,
+                    n_lags=3,
+                    weekly_seasonality=False,
+                    daily_seasonality=False,
+                    epochs=EPOCHS,
+                    batch_size=BATCH_SIZE,
+            )
+            m = m.add_lagged_regressors(names=cols)
+            metrics_df = m.fit(df1, freq="D", validate_each_epoch=True)
+            future = m.make_future_dataframe(df1, n_historic_predictions=365)
+            forecast = m.predict(future)
```

Comment:
here, add a check that the correct A, B, C components are present in the forecast dataframe